tools/grammars/compiler/data.go

```diff
@@ -46,6 +48,7 @@ var GrammarsInNonStdPath = map[string]bool{
 	"hy.tmLanguage":           true,
 	"abl-tmlanguage":          true,
 	"avro.tmLanguage":         true,
+	"grammar":                 true,
```

Comment:
This should be the name of the grammar as it appears in the filesystem:

```suggestion
	"ionide-fsgrammar":                 true,
```

Reply:
I tried this. but it is a bit more complicated. Here we are looking for the the name of (sub)directory inside the submodule, which has name other than `grammars/` (plural). I have sent a PR upstream, so subsequently we could remove this workaround: https://github.com/ionide/ionide-fsgrammar/pull/156.

Reply:
`GrammarsInNonStdPath` is used here: https://github.com/github/linguist/blob/5ff272d46e519914ed823d2cc6f670b205a3ea5d/tools/grammars/compiler/loader.go#L125 the map needs the name of the directory on the filesystem, that has the grammar file. in this case name of the directory is `grammar/`.

Reply:
Upstream PR (that moves grammer files from `grammar/` to `grammars/`) has been merged. I have updated the submodule and rebased.